### DISCUSSION TITLE
Using FontAwesome

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#9144] [Q&A]
[_From a Discord discussion with SirNalon_](https://discord.com/channels/372137812037730304/550416151172087808/990784486885195786):

My team and I have been developing an application using uno-platform and I have googled, tried the supplied documentation and just cannot get FontAwesome to work globally (or at all) for my WinUI app. I do not have the font installed on my system (but this is on purpose as i want it to be used from the Assets folder) any suggestions or help would be amazing.

![image](https://user-images.githubusercontent.com/5839577/175851055-1f152ec7-d095-4f40-831d-136110d73406.png)

This is how i'm currently attempting to call it within the app.cs within my Shared Project

![image](https://user-images.githubusercontent.com/5839577/175851070-286168ae-fe47-4d06-a1d3-8c651007bf9e.png)

**Comment by jeromelaban** [comment]
Thanks for the question! Could you specify which target you want to run with? Is it WinAppSDK on Windows? Is it another Uno target?

**Reply by nalonau** [reply]
Hey Jerome,

Thanks for opening this discussion! I am targeting WinAppSDK on windows. I will eventually want to target mobile, but my main goal is to get windows working first

**Reply by nalonau** [reply]
incase this helps this is the desktop.csproj file
![image](https://user-images.githubusercontent.com/66582678/175855753-af6dc700-ad82-4943-9a4d-40eb1d9fc2f3.png)

**Reply by jeromelaban** [reply]
Thanks. As this is a WinAppSDK issue, most the windows documentation will help you, or [blogs like this one](https://blog.mzikmund.com/2017/01/using-custom-fonts-in-cs-uwp-apps). 

Are you able to use other types of fonts or is it just this one?

**Reply by nalonau** [reply]
Thanks for the link, I will give this a go and see if I have any luck. I've tried a lot of examples which don't seem to work (I just get the square _missing_ icon.

I haven't tried a different icon, so i might do that as-well then report my findings. I noticed (and I'm not sure if this is possible) but I did try using fontawesome on a MAUI project and they had a really easy way of adding a new font and this did work (but I understand this is a different project type). You basically register the fonts using the builder during app start-up. Is this something that may eventually be able to come over to Uno? https://docs.microsoft.com/en-us/dotnet/maui/user-interface/fonts

**Reply by jeromelaban** [reply]
If you got the font working on MAUI on Windows, this likely means that the registration of the font in your Windows project is not defined properly (as MAUI uses WinUI directly). The problem here, unless this changed, is that this font is not available publicly, so it's not possible to help you there.

**Reply by nalonau** [reply]
That makes sense, so i might try a few things like installing the font and see if I can at least get it working that way. Ive noticed recently the 'Unpackaged' mode has become available which will give me more options to use a 3rd party installer that could install the font directly to the system.

Really appreciate the help!
